ontology_processing/graph_creation/make_graph_class.py

```diff
@@ -0,0 +1,652 @@
+import pickle
+import argparse
+import networkx as nx
+import pandas as pd
+import validators
+
+import owlready2
+from owlready2 import sync_reasoner
+from collections import OrderedDict
+
+try:
+    from graph_creation.ontology_processing_utils import (
+        give_alias,
+        save_test_ontology_to_json,
+        save_graph_to_pickle,
+        get_valid_test_ont,
+        get_non_test_ont,
+        remove_non_test_nodes,
+        get_test_ontology,
+        get_source_types,
+        solution_sources,
+        listify,
+        union_subgraph,
+        custom_bfs,
+    )
+except ImportError:
+    from ontology_processing.graph_creation.ontology_processing_utils import (
+        give_alias,
+        save_test_ontology_to_json,
+        save_graph_to_pickle,
+        get_valid_test_ont,
+        get_non_test_ont,
+        remove_non_test_nodes,
+        get_test_ontology,
+        get_source_types,
+        solution_sources,
+        listify,
+        union_subgraph,
+        custom_bfs,
+    )
+
+# Set a lower JVM memory limit
+owlready2.reasoning.JAVA_MEMORY = 500
```

Comment:
This is just for the “automated reasoning” part. Could move this as part of the automated reasoning function?

Reply:
Done.